sklearn/feature_selection/_rfe.py

```diff
@@ -184,14 +199,24 @@ def _fit(self, X, y, step_score=None):
             estimator.fit(X[:, features], y)
 
             # Get coefs
-            if hasattr(estimator, 'coef_'):
-                coefs = estimator.coef_
+            if callable(self.importance_getter):
```

Comment:
This could be neater. Define a method `_get_importances_auto` that handles the auto case, then this can look like:
```py
getter = self.importance_getter
if isinstance(getter, str):
    if getter == 'auto':
        getter = self._get_importances_auto
    else:
        getter = attrgetter(getter)
if not callable(getter):
    raise ValueError
return getter(estimator)
```

Reply:
This looks a lot more readable. Thanks, with minor changes implemented this.